lib/foreman_tasks_core/task_launcher/batch.rb

```diff
@@ -7,53 +7,14 @@ class ParentAction < ::Dynflow::Action
       # { task_id => { :action_class => Klass, :input => input } }
       def plan(launcher, input_hash)
         launcher.launch_children(self, input_hash)
-        sequence do
```

Comment:
@adamruzicka I've removed this part of https://github.com/theforeman/foreman-tasks/pull/385/commits/d2af822a59439f848748e72aebc6c97224040870#diff-854a78e251be71cf44b31b98e342ee22R10, as it was changing the end2end behaviour of the ssh provider (so no hosts would be marked as finished until the last host from the batch finishes) and could look like a regression.

Reply:
Also, it was not working when there were some failed hosts, as the task failed with `Stop` rescue_scenario